lib/page.js

```diff
@@ -98,13 +98,20 @@ class PDFPage {
     });
 
     // The page dictionary
-    this.dictionary = this.document.ref({
+    const pageDictionary = {
       Type: 'Page',
       Parent: this.document._root.data.Pages,
       MediaBox: [0, 0, this.width, this.height],
       Contents: this.content,
-      Resources: this.resources
-    });
+      Resources: this.resources,
+    };
+
+    // Set tab order if document is tagged for accessibility.
+    if (this.document.getMarkInfoDictionary().data.Marked === true) {
```

Comment:
This won't quite have the desired effect, as calling `getMarkInfoDictionary` will create that dictionary on demand, and I don't think we should do that for all PDFs. I think it would be better to add a `hasMarkInfoDictionary` method to check for its existence.

I'm also not sure this change should be restricted to Tagged PDF by checking for `.Marked === true`. I think it may make sense to set `Tabs` to `S` whenever there is structure, whether it's a Tagged PDF or not. This causes a race condition, because the first page may be created before structure is added for a non-tagged PDF.

So I would suggest moving the logic to add `Tabs` into the page `end()` method (preferably by adding a private method with the logic). Note that it's OK to mutate `this.dictionary` until `this.dictionary.end()` is called, cf. `get structParentTreeKey()`. I think in that location, it would make sense to add the entry:
- only if the document `MarkInfo` dictionary exists (but regardless of whether `Marked` is `true`)
- only if there is no `Tabs` entry already in the page dictionary (in case at some point we have an alternative way to set it, e.g. via options to the constructor, or some other method, or a user wants to "hack it" themselves)

Reply:
great feedback! happy to work on these changes.

Reply:
@insightfuls thanks for this review! I've updated the approach as suggested and added a test to attempt to ensure that the taborder is not set when the document is not marked, let me know what you think.